lib/iaas/GcpClient.js

```diff
@@ -0,0 +1,200 @@
+'use strict';
+const _ = require('lodash');
+const Promise = require('bluebird');
+const GcpStorage = require('@google-cloud/storage');
+const logger = require('../logger');
+const errors = require('../errors');
+const utils = require('../utils');
+const ComputeClient = require('./ComputeClient');
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class GcpClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.constructor.validateParams(_.chain(this.settings).value());
+    this.storageClient = this.constructor.createStorageClient(_
+      .chain(this.settings)
+      .omit('name')
+      .set('provider', this.provider)
+      .value()
+    );
+    this.computeClient = this.constructor.createComputeClient(_
+      .chain(this.settings)
+      .set('provider', this.provider)
+      .value()
+    );
+  }
+
+  getContainer(container) {
+    if (arguments.length < 1) {
+      container = this.containerName;
+    }
+
+    return this.storageClient
+      .bucket(container)
+      .get();
+  }
+
+  list(container, options) {
+    if (arguments.length < 2) {
+      options = container;
+      container = this.containerName;
+    }
+    const prefix = options ? options.prefix : null;
+    const queryOptions = {};
+    if (prefix) {
+      queryOptions.prefix = prefix;
+    }
+    queryOptions.autoPaginate = (options && options.autoPaginate) ? options.autoPaginate : true;
+
+    return Promise.try(() => {
```

Comment:
if there is a single statement the '{' and 'return' can be avoided. Also have you tried promisifying the functions of the lib ? Then may be you dont have to wrap all methods with a Promise.try().

Reply:
Will remove '{' and return.
The functions are already promisified but it's normal promise and not the bluebird promise. Hence wrapped around try().